### PR TITLE
refactor: remove tox py3X tests

### PR DESCRIPTION
**Pull request by CalvoM**
## Why is this needed?

Removing the py3X tests in tox.ini since we are moving testing to building in the specified 
series.

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] Changes here need to be documented, and this was done in: 

## Does this PR require extra reviews?

 - [ ] Yes
 - [x] No

### CODE REVIEW

**Comment by renanrodrigo** [comment]
You could also remove `tools/setup_pyenv.sh` as we don't really need pyenv anymore

File: tox.ini [hunk 1]
```diff
@@ -61,9 +32,7 @@ setenv =
 commands =
     test: py.test --junitxml=pytest_results.xml {posargs:--cov uaclient uaclient}
     flake8: flake8 uaclient lib setup.py features
-    mypy: mypy --python-version 3.6 uaclient/ features/ lib/
```
[the remark this answers is not in the document]
Comment (orndorffgrant) [reply tox.ini:64]: Can we add mypy to the build as well? Then I would feel better dropping old-version mypy here.

[the remark this answers is not in the document]
Comment (renanrodrigo) [reply tox.ini:64]: +1 that would be good

[the remark this answers is not in the document]
Comment (CalvoM) [reply tox.ini:64]: after testing, it seems we will need to install the `type-*` packages as well. I am not sure it is something we need to add to the packages. I prefer returning the mypy commands back to `tox.ini`

[the remark this answers is not in the document]
Comment (renanrodrigo) [reply tox.ini:64]: After a quick discussion, we think we can totally remove the older mypy testing, as you have done in the fisrt iteration, @CalvoM ; sorry for the digression here, we just wanted to make sure this wouldn't hurt... and it really won't.

**Review by renanrodrigo — APPROVED** [review]
Great, thanks much!

**Pull request merged**